cardano-tracer/src/Cardano/Tracer/Handlers/Metrics/Monitoring.hs

```diff
@@ -62,10 +92,65 @@ runMonitoringServer tracerEnv (Endpoint listHost listPort, monitorEP) = do
             , UI.jsSSLChainCert = False
             }
       }
+#else
+runMonitoringServer tracerEnv (endpoint@(Endpoint listHost listPort), monitorEP) = do
+  -- Pause to prevent collision between "Listening"-notifications from servers.
+  sleep 0.2
+  traceWith (teTracer tracerEnv) TracerStartedMonitoring
+    { ttMonitoringEndpoint = endpoint
+    , ttMonitoringType     = "list"
+    }
+  simpleHttpServe config do
+    route 
+      [ ("/", renderEkg)
+      ]
+ where
+  TracerEnv{teConnectedNodes} = tracerEnv
 
--- | We have to keep an id of the node as well as thread id of currently launched EKG server.
-type CurrentEKGServer = TMVar (NodeId, ThreadId)
+  config :: Config Snap ()
+  config = defaultConfig
+    & setErrorLog ConfigNoLog
+    & setAccessLog ConfigNoLog
+    & setBind (encodeUtf8 (T.pack listHost))
+    & setPort (fromIntegral listPort)
+
+  renderEkg :: Snap ()
+  renderEkg = do
+    nodes <- liftIO $ S.toList <$> readTVarIO teConnectedNodes
 
+    -- HACK
+    case nodes of
+      [] ->
+        pure ()
+      nodeId:_nodes -> liftIO do
+        restartEKGServer tracerEnv nodeId monitorEP currentServerHack
+    blaze do
+      docTypeHtml do
+        ekgHtml monitorEP nodes
+
+{-# NOINLINE currentServerHack #-}
+currentServerHack :: CurrentEKGServer
+currentServerHack = unsafePerformIO newEmptyTMVarIO
```

Comment:
Please provide a short comment why this hack is currently necessary, i.e. in what way it relates to the ability to safely restart an EKG server.